File: frontend/src/main/models/Resume.ts
export interface Resume {
    id: string;
    name: string;
    uploadDate: string;
    status: 'processing' | 'completed' | 'error';
    atsScore?: number;
    suggestions?: number;
    fileSize?: number;
    fileType?: string;
}

export interface ResumeStats {
    total: number;
    completed: number;
    processing: number;
    avgScore: number;
}

File: backend/src/main/java/com/sammeydan/resumeanalyzer/backend/model/Resume.java
package com.sammeydan.resumeanalyzer.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Resume {
    private String id;
    private String name;
    private LocalDateTime uploadDate;
    private String status; // 'processing', 'completed', 'error'
    private Integer atsScore;
    private Integer suggestions;
    private Long fileSize;
    private String fileType;
}
